components/script/dom/htmloptionscollection.rs

```diff
@@ -0,0 +1,194 @@
+            let name = QualName::new(ns!(html), name_atom.clone());
+            let element = Element::create(name, None, document.r(), ElementCreator::ScriptCreated);
+            let node = element.r().upcast::<Node>();
+            try!(Node::pre_insert(node, root.r(), None));
+        };
+        Ok(())
+    }
+}
+
+impl HTMLOptionsCollectionMethods for HTMLOptionsCollection {
+    // FIXME: This shouldn't need to be implemented here since HTMLCollection (the parent of
+    // HTMLOptionsCollection) implements NamedGetter.
+    // https://github.com/servo/servo/issues/5875
+    //
+    // https://dom.spec.whatwg.org/#dom-htmlcollection-nameditem
+    fn NamedGetter(&self, name: DOMString) -> Option<Root<Element>> {
+        self.collection.NamedItem(name)
+    }
+
+    // https://heycam.github.io/webidl/#dfn-supported-property-names
+    fn SupportedPropertyNames(&self) -> Vec<DOMString> {
+        self.collection.SupportedPropertyNames()
+    }
+
+    // FIXME: This shouldn't need to be implemented here since HTMLCollection (the parent of
+    // HTMLOptionsCollection) implements IndexedGetter.
+    // https://github.com/servo/servo/issues/5875
+    //
+    // https://dom.spec.whatwg.org/#dom-htmlcollection-item
+    fn IndexedGetter(&self, index: u32) -> Option<Root<Element>> {
+        self.collection.IndexedGetter(index)
+    }
+
+    // https://html.spec.whatwg.org/multipage/#dom-htmloptionscollection-setter
+    fn IndexedSetter(&self, index: u32, value: Option<&HTMLOptionElement>) -> ErrorResult {
+        if let Some(value) = value {
+            // Step 2
+            let length = self.collection.Length();
+
+            // Step 3
+            let n = index as i32 - length as i32;
+
+            // Step 4
+            if n > 0 {
+                try!(self.add_new_elements(n as u32));
+            }
+
+            // Step 5
+            let node = value.upcast::<Node>();
+            let root = self.root.get();
+            if n >= 0 {
+                Node::pre_insert(node, root.r(), None).map(|_| ())
+            } else {
+                let child = self.collection.IndexedGetter(index).unwrap();
+                let child_node = child.r().upcast::<Node>();
+
+                root.r().ReplaceChild(node, child_node).map(|_| ())
+            }
+        } else {
+            // Step 1
+            self.Remove(index as i32)
+        }
+    }
+
+    // https://html.spec.whatwg.org/multipage/#dom-htmloptionscollection-length
+    fn Length(&self) -> u32 {
+        self.collection.Length()
+    }
+
+    // https://html.spec.whatwg.org/multipage/#dom-htmloptionscollection-length
+    fn SetLength(&self, length: u32) {
+        let current_length = self.collection.Length();
+        let delta = length as i32 - current_length as i32;
+        if delta < 0 {
+            // new length is lower - deleting last option elements
+            for index in (length..current_length).rev() {
+                self.Remove(index as i32).unwrap()
```

Comment:
Unwrapping is not fine here, since it can fail. If you don't care about the result of removing, you can do `let _ = self.Remove(index as i32);`